bosh-director/spec/unit/deployment_plan/instance_plan_spec.rb

```diff
@@ -119,6 +119,36 @@ module Bosh::Director::DeploymentPlan
           instance_plan.networks_changed?
         end
 
+        context 'when dns_record_name exists in network_settings' do
+          let(:network_settings) do
+            {
+                'existing-network' =>{
+                    'type' => 'dynamic',
+                    'cloud_properties' =>{},
+                    'dns_record_name' => '0.job-1.my-network.deployment.bosh',
+                    'dns' => '10.0.0.1',
+                }
+            }
+          end
+          
+          it 'should ignore dns_record_name when comparing old and new network_settings' do
+            new_network_settings = {
+                'existing-network' =>{
+                    'type' => 'dynamic',
+                    'cloud_properties' =>{},
+                    'dns' => '10.0.0.1',
+                }
+            }
+
+            allow(logger).to receive(:debug)
+            expect(logger).to_not receive(:debug).with(
+                "networks_changed? network settings changed FROM: #{network_settings} TO: #{new_network_settings} on instance #{instance_plan.existing_instance}"
+            )
+
+            instance_plan.networks_changed?
```

Comment:
Since it was already merged at one point and our existing tests were a bit
weird to begin with, maybe lets just re-merge and we clean up whatever we
need to. Seems easier and more helpful than back-and-forth on these details.


On Wednesday, September 7, 2016, Ming Xiao <notifications@github.com> wrote:

> In bosh-director/spec/unit/deployment_plan/instance_plan_spec.rb
> <https://github.com/cloudfoundry/bosh/pull/1411#discussion_r77882636>:
>
> > +
> > +          it 'should ignore dns_record_name when comparing old and new network_settings' do
> > +            new_network_settings = {
> > +                'existing-network' =>{
> > +                    'type' => 'dynamic',
> > +                    'cloud_properties' =>{},
> > +                    'dns' => '10.0.0.1',
> > +                }
> > +            }
> > +
> > +            allow(logger).to receive(:debug)
> > +            expect(logger).to_not receive(:debug).with(
> > +                "networks_changed? network settings changed FROM: #{network_settings} TO: #{new_network_settings} on instance #{instance_plan.existing_instance}"
> > +            )
> > +
> > +            instance_plan.networks_changed?
>
> consulted with @dpb587 <https://github.com/dpb587> , and we think that
> the result should be false. Since the network did not change and we are
> now ignoring dns_record_name, networks_changed? should return false
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/cloudfoundry/bosh/pull/1411/files/533d03117c1a48c1bdad56bd197b0c7bde1fb15b#r77882636>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/APYVcFqPLSW8r2UfVrpI6QHbZzSM2XUPks5qnwhogaJpZM4JrxLq>
> .
>